### PR TITLE
Fix missing avatar for show current profiles

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes https://github.com/vector-im/element-web/issues/23733

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix missing avatar for show current profiles ([\#9563](https://github.com/matrix-org/matrix-react-sdk/pull/9563)). Fixes vector-im/element-web#23733.

### CODE REVIEW

**Review by t3chguy — CHANGES_REQUESTED** [review]
Given this is a fix for a regression, can we get a test which asserts that it doesn't break again please?

**Comment by gsouquet** [comment]
@t3chguy I have not been able to reproduce those steps in testing just yet. I can try on for longer but haven't had any luck yet :/

**Comment by t3chguy** [comment]
I can reproduce after enabling

image

and refreshing

image

The test doesn't have to be a cypress test, just a minimal test case which would have caught this edge is plenty

File: src/components/views/avatars/MemberAvatar.tsx [hunk 1]
```diff
@@ -59,25 +59,37 @@ export default function MemberAvatar({
...
-    if (member?.name) {
-        if (member.getMxcAvatarUrl()) {
-            imageUrl = mediaFromMxc(member.getMxcAvatarUrl() ?? "").getThumbnailOfSourceHttp(
-                width,
-                height,
-                resizeMethod,
-            );
+    const [name, setName] = useState<string>();
+    const [imageUrl, setImageUrl] = useState<string>();
+    const [title, setTitle] = useState<string>();
```
[the remark this answers is not in the document]
Comment (gsouquet) [reply src/components/views/avatars/MemberAvatar.tsx:64]: Unfortunately using `useMemo` does not fix the bug here. I'll stick with `useEffect`

**Comment by robintown** [comment]
@gsouquet I think I've managed to fix this in a simpler way: https://github.com/matrix-org/matrix-react-sdk/commit/51932493b24421dc2b7573ac3287a549319d1879

The issue seems to be that pills are rendered in isolated component trees, and so don't have a room in their RoomContext to work with. Falling back to the prop member in this case restores the previous behavior.

**Review by t3chguy — APPROVED** [review]
Looks sane to me!

**Pull request merged**